### ISSUE TITLE
2020-01-06 M4SS AE generates NA value for all records except the first when using nominal time introduced with cde2560 / dce4ed8

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#147]
### Issue

When testing tc018 M4SS with nominal time points, only the first record has a non-NA value given as an output. This was introduced with the cde2560 / dce4ed8 commits. This appears to occur for all M4SS testcases. This does not seem to occur for M4SD.

#### Output as of commit dce4ed8

```r
> table(r$AE.12.00, useNA = "always")
6328000    <NA> 
      1      44 

> head(r, 3)
       SDEID AMT.12.00 AMT.24.00 AE.12.00 AE.24.00        AE   AETAU1 AETAUPT1   MAXRATE TMAXRATE  MAXRATE1 TMAXRATE1 RATELAST1 MIDPTLAST1 TLAG KEL
1   43456658   6328000   3718000  6328000  3718000  10046000 10046000 33486667  527333.3        6  527333.3         6  309833.3         18    6  NA
2   98862581  31625000  12847840       NA       NA  44472840       NA       NA 2635416.7        6 2635416.7         6 1070653.3         18    6  NA
3  102652097  79376000  27738000       NA       NA 107114000       NA       NA 6614666.7        6 6614666.7         6 2311500.0         18    6  NA
```

#### Output with historic commit fcf5096

```r
> head(r, 3)
       SDEID AMT.12.00 AMT.24.00 AE.12.00 AE.24.00        AE    AETAU1 AETAUPT1   MAXRATE TMAXRATE  MAXRATE1 TMAXRATE1 RATELAST1 MIDPTLAST1 TLAG KEL
1   43456658   6328000   3718000  6328000  3718000  10046000  10046000 33486667  527333.3        6  527333.3         6  309833.3         18    6  NA
2   98862581  31625000  12847840 31625000 12847840  44472840  44472840 22236420 2635416.7        6 2635416.7         6 1070653.3         18    6  NA
3  102652097  79376000  27738000 79376000 27738000 107114000 107114000 53557000 6614666.7        6 6614666.7         6 2311500.0         18    6  NA
```

**Comment by rudraya** [comment]
This issue should be resolved as of 1e16a39. Please let me know if it still occurs

**Comment by jhhughes256** [comment]
This issue no longer occurs as of commit e681d9c / 1e16a39. Example output below.

```r
> head(r_old, 3)
       SDEID AMT.12.00 AMT.24.00 AE.12.00 AE.24.00        AE   AETAU1 AETAUPT1   MAXRATE TMAXRATE  MAXRATE1 TMAXRATE1 RATELAST1 MIDPTLAST1
1   43456658   6328000   3718000  6328000  3718000  10046000 10046000 33486667  527333.3        6  527333.3         6  309833.3         18
2   98862581  31625000  12847840       NA       NA  44472840       NA       NA 2635416.7        6 2635416.7         6 1070653.3         18
3  102652097  79376000  27738000       NA       NA 107114000       NA       NA 6614666.7        6 6614666.7         6 2311500.0         18
> head(r_new, 3)
       SDEID AMT.12.00 AMT.24.00 AE.12.00  AE.24.00        AE    AETAU1  AETAUPT1   MAXRATE TMAXRATE  MAXRATE1 TMAXRATE1 RATELAST1
1   43456658   6328000   3718000  6328000  10046000  10046000  16374000  54580000  527333.3        6  527333.3         6  309833.3
2   98862581  31625000  12847840 31625000  44472840  44472840  76097840  38048920 2635416.7        6 2635416.7         6 1070653.3
3  102652097  79376000  27738000 79376000 107114000 107114000 186490000  93245000 6614666.7        6 6614666.7         6 2311500.0
```
